### PR TITLE
Add pooch as our data manager

### PR DESCRIPTION
**Pull request by jni**
## Description

This PR adds pooch as a new dependency and adds the cells.tif dataset from our tutorials as the single external dataset for now. In the future it enables scikit-image to provide much more diverse data e.g. in tutorials.

Supersedes #3945. 

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by hmaarrfk** [comment]
I think this needs a rebase... thats where i got stuck.

**Comment by emmanuelle** [comment]
@hmaarrfk CI is failing for something introduced in this PR (which was not in your original one). @jni you can also take it from #4484 where I merged master (since we squashing I guess it's ok to merge instead of rebasing?).

File: skimage/data/setup.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+from ._registry import legacy_datasets
```
Comment (emmanuelle) [inline skimage/data/setup.py]: this line makes CI fail, please remove the relative import

Comment (jni) [reply skimage/data/setup.py]: @emmanuelle do you have any idea why this fails, though?

Comment (hmaarrfk) [reply skimage/data/setup.py]: I think setup is run prior to install

Comment (emmanuelle) [reply skimage/data/setup.py]: ah ok interesting, thanks @hmaarrfk. But removing the relative import also fails the CI... Should this be `from skimage.data._registry import legacy_datasets`?

**Pull request closed without merging**